content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,80 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
 
 # Automatic Sharding
 
-> *Note:* Automatic sharding is not yet part of this specification.
-This section merely serves as an outlook.
-A specification of automatic sharding will be added to this document in a future version.
+Autosharding select shards automatically and is the default behaviour for shard choice but opting out is always possible.
+Shards (pubsub topics) not longer have to be chosen, they are instead computed from content topics with the procedure below.
+
+## Rendezvous Hashing
+
+Also known as the Highest Random Weight (HRW) method, has many properties useful for shard selection.
+- Distributed agreement. Local only computation without communication.
+- Load balancing. Content topics are spread as equally as possible to shards.
+- Minimal disruption. Content topics switch shards infrequently.
+- Low overhead. Easy to compute.
+
+### Algorithm
+
+For each shard,
+hash using Sha2-256 the concatenation of
+the content topic `application` field (N UTF-8 bytes),
+`version` (N UTF-8 bytes),
+the cluster index (2 bytes) and
+the shard index (2 bytes)
+take the first 64 bits of the hash,
+divided by 2^64,
+compute the natural logarithm of this number then
+take the negative of the weight (default 1.0) divided by it.
+Finally, sort the values and pick the shard with the highest one.
+
+## Content Topic Prefixes
```

Comment:
Perhaps just title this `Content Topic Format for Autosharding` and have a link back to the general content topic format recommendation in https://rfc.vac.dev/spec/23/#content-topic-format.
This will allow this section to be an easy reference for the format and not just the optional prefixes. When I go this section I want to know:
(1) quick reference of the content topic format I should use
(2) more detail about optional prefixes